Automattic/wp-calypso
# Components: Remove global fieldset styles

#### Changes proposed in this Pull Request

* Remove global form fieldset styles in favor of more specific component-level styling. See #45259.

The following use cases of fieldsets are consciously ignored:

* `apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-template-modal/index.js` - uses `@wordpress/components` and not Calypso ones.
* `calypso/client/landing/gutenboarding/components/signup-form/index.tsx` - Gutenboarding uses `@wordpress/components` and not Calypso ones

#### Testing instructions

Test the following against production and ensure that they match stylistically and functionally:

* Editing a comment in the full post view in the Reader. To get there, you will need to enable the `showModerationTools` prop in `<PostCommentList />` with your browser devtools or manually in your local Calypso. Go to a post with comments in the Reader. Click the Edit button below a comment, and compare the fieldset around the comment content field.
* Posting a new comment in the full post view in the Reader. Go to a post with comments in the Reader. Scroll down to the comment form at the bottom of the comments thread, and compare the fieldset around the comment content field.
* Community translator - to test it, you need to enable the `i18n/community-translator` flag, and make sure `isCommunityTranslatorEnabled()` returns true. Then, compare the fieldset that wraps the translatable field. 
* `/devdocs/design/date-range` - fieldset around date inputs when you click the first example. 
* inserting a contact form and editing a field in the classic calypso editor. observe the fieldset around the field editing fields.
* Editing a custom nameservers row for a domain bought from WP.com where you don't use WP.com's nameservers but use custom ones.
* `/domains/add/site-redirect/:site` where `:site` is a WP.com site.
* `/marketing/sharing-buttons/:site`, the entire page, there are multiple fieldsets on that page.
* `/marketing/traffic/:site` where `:site` is a site with Premium or Business plan; scroll down to the Google Analytics card and compare the 3 fieldsets there.
* fieldsets in the editor drawer of the classic Calypso editor. there is one in the Discussion section, and one in the Sharing section.
* `/media/:site` - the media edit popup, the fieldsets on the right.
* Areas that contain fieldsets in Calypso - settings, domains, me, etc.

#### Notes

Lint errors are expected and unrelated.

See #45259 for rationale on why we're working on this.
